### ISSUE TITLE
When finetune the model, how do I use different learning rate for different layers?

### ISSUE THREAD

**Issue by coolmatt1024** [keras-team/keras#8850]
I was using resnet-50 to do saliency detection. When I used learning rate 1e-3, the validation loss started to increase just after one epoch; when I reduce the learning rate to 1e-4 or 3e-4,or 5e-4, the validation rate decreased but the final result was worse than  1e-3. I think the learning rate was too high for the resnet layers, but the learning rate is too slow for other layers.

So, can I use different learning rates for different layers? Could you give some tips?

**Comment by gabrieldemarmiesse** [comment]
Hi! 
It's impossible with the current way the framework is made.
 But if you're willing to modify the keras code, you can look up the file optimizer.py. 
Let's say you want to add a factor when the weights are updated with plain SGD. In line 168, there is a loop over all the weights of the network and this is where the gradients are multiplied by the learning rate.
If you're willing to do it, this would be the place to start. But i'm not sure what the most beautiful implementation of this would look like. 

Hope this helps.
